invokeai/frontend/web/src/features/system/components/SettingsModal/SettingsClearIntermediates.tsx

```diff
@@ -0,0 +1,56 @@
+import { RootState } from 'app/store/store';
+import { useAppDispatch, useAppSelector } from 'app/store/storeHooks';
+import { useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { StyledFlex } from './SettingsModal';
+import { Heading, Text } from '@chakra-ui/react';
+import IAIButton from '../../../../common/components/IAIButton';
+import { useClearIntermediatesMutation } from '../../../../services/api/endpoints/images';
+import { addToast } from '../../store/systemSlice';
+
+export default function SettingsClearIntermediates() {
+  const dispatch = useAppDispatch();
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const [clearIntermediates, { isLoading: isLoadingClearIntermediates, data }] =
+    useClearIntermediatesMutation();
+
+  const handleClickClearIntermediates = useCallback(() => {
+    clearIntermediates({});
+  }, [clearIntermediates]);
+
+  useEffect(() => {
+    if (data >= 0) {
+      dispatch(
+        addToast({
+          title:
+            data === 0
+              ? `No intermediates to clear`
+              : `Successfully cleared ${data} intermediates`,
+          status: 'info',
+        })
+      );
+      if (data < 100) {
+        setIsDisabled(true);
+      }
+    }
+  }, [data, dispatch]);
```

Comment:
I don't like this being in a `useEffect()`, but I don't not like it enough to not approve this.

Reply:
I put it all in the unwrap and it seems to work well. TIL, thanks!